### ISSUE TITLE
'resign' action requires entitlements, which is documented as an optional parameter

### ISSUE THREAD

**Issue by zachlipton** [fastlane/fastlane#1471]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description

I invoke 'resign' with the arguments: 
```
resign(signing_identity: 'iPhone Distribution: Viggle, Inc. (9K99382NDB)',
    			 provisioning_profile: 'AppStore_net.functionxinc.oda-ota.mobileprovision')
```
This worked fine previously, but now fails with a crash (report #640af042151c8256c9f19b8a740a98b2) :
`/Users/zach/.rvm/gems/ruby-2.2.1@global/gems/sigh-1.4.0/lib/sigh/resign.rb:14:in `resign': [!] wrong number of arguments (4 for 5) (ArgumentError)`

It appears that the 'entitlements' parameter (added in this commit: https://github.com/fastlane/fastlane/commit/8a29d94be2b6bc552462a941ed6cdb39f6a8c24a) is supposed to be optional, but the resign action expects a value to be present anyway. This is not documented anywhere.

##### Complete output when running fastlane, including the stack trace and command used

```
fastlane testflight
...
[16:41:57]: --------------------
[16:41:57]: --- Step: resign ---
[16:41:57]: --------------------
[16:41:57]: Variable Dump:
[16:41:57]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios testflight", :SIGH_PROFILE_PATH=>"/Users/zach/git/viggle/oda-ios/AppStore_net.functionxinc.oda-ota.mobileprovision", :SIGH_PROFILE_PATHS=>["/Users/zach/git/viggle/oda-ios/AppStore_net.functionxinc.oda-ota.mobileprovision"], :SIGH_UDID=>"10f6dbef-2a19-46b0-8765-ec37034aa924", :SIGH_PROFILE_TYPE=>"app-store", :IPA_OUTPUT_PATH=>"/Users/zach/git/viggle/oda-ios/Viggle.ipa", :DSYM_OUTPUT_PATH=>"/Users/zach/git/viggle/oda-ios/Viggle.app.dSYM.zip", :XCODEBUILD_ARCHIVE=>"/Users/zach/Library/Developer/Xcode/Archives/2016-03-03/Viggle 2016-03-03 16.39.14.xcarchive"}
[16:41:57]: wrong number of arguments (4 for 5)

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | cocoapods                           | 50          |
| 4    | sigh                                | 19          |
| 5    | gym                                 | 171         |
| 6    | resign                              | 0           |
+------+-------------------------------------+-------------+

[16:41:57]: fastlane finished with errors
Successfully submitted a crash report. If this is a problem with one of the tools specifically,
please submit an issue on GitHub and attach the following number to it: '640af042151c8256c9f19b8a740a98b2'
The crash report has been stored locally '/tmp/sentry_640af042151c8256c9f19b8a740a98b2.json'
/Users/zach/.rvm/gems/ruby-2.2.1@global/gems/sigh-1.4.0/lib/sigh/resign.rb:14:in `resign': [!] wrong number of arguments (4 for 5) (ArgumentError)
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/actions/resign.rb:9:in `run'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/runner.rb:143:in `block (2 levels) in execute_action'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/actions/actions_helper.rb:35:in `execute_action'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/runner.rb:129:in `block in execute_action'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/runner.rb:128:in `chdir'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/runner.rb:128:in `execute_action'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/fast_file.rb:149:in `method_missing'
	from Fastfile:68:in `block (2 levels) in parsing_binding'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/lane.rb:36:in `call'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/lane.rb:36:in `call'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/runner.rb:52:in `block in execute'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/runner.rb:46:in `chdir'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/runner.rb:46:in `execute'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/lane_manager.rb:46:in `cruise_lane'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/lib/fastlane/command_line_handler.rb:30:in `handle'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/bin/fastlane:38:in `block (2 levels) in run'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane_core-0.36.8/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/bin/fastlane:156:in `run'
	from /Users/zach/.rvm/gems/ruby-2.2.1@global/gems/fastlane-1.65.0/bin/fastlane:162:in `<top (required)>'
	from /Users/zach/.rvm/gems/ruby-2.2.1/bin/fastlane:23:in `load'
	from /Users/zach/.rvm/gems/ruby-2.2.1/bin/fastlane:23:in `<main>'
	from /Users/zach/.rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `eval'
	from /Users/zach/.rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `<main>'

```

### Configuration Files

Please copy the complete content of your `Fastfile` and any other configuration files you use below:

Fastfile:
```ruby
fastlane_version "1.35.0"

default_platform :ios


platform :ios do
  before_all do
    cocoapods
  end

  desc "Runs all the tests"
  lane :test do
    # snapshot
  end
  
  desc "Do hockey app build"
  lane :hockeyapp do
  	sigh(provisioning_name: "ViggleAdHoc",
    		 adhoc: true)
    gym(scheme: "oda-ios_ci",
    		workspace: "oda-ios.xcworkspace",
    		configuration: "AdHoc",
    		clean: true,
    		export_method: 'ad-hoc',
    		use_legacy_build_api: true,
    		codesigning_identity: 'iPhone Distribution: Viggle, Inc. (9K99382NDB)')
    hockey(
  		api_token: 'XXX',
  		ipa: './Viggle.ipa',
  		notes: git_branch
		) 
  end

	desc "Do TestFlight/Release build"
	lane :testflight do
		sigh(provisioning_name: "ViggleAppStore")
		ENV["PROFILE_UDID"] = lane_context[SharedValues::SIGH_UDID]
    gym(scheme: "oda-ios_fx",
    		workspace: "oda-ios.xcworkspace",
    		configuration: "AppStore Distribution",
    		clean: true,
    		export_method: 'app-store',
    		use_legacy_build_api: true,
    		codesigning_identity: 'iPhone Distribution: Viggle, Inc. (9K99382NDB)')
    resign(signing_identity: 'iPhone Distribution: Viggle, Inc. (9K99382NDB)',
    			 provisioning_profile: 'AppStore_net.functionxinc.oda-ota.mobileprovision')
    testflight(skip_submission: true) 
	end
```

### Environment

fastlane version (fastlane -v): fastlane 1.65.0

**Comment by asfalcone** [comment]
@zachlipton Thanks for reporting. It looks like the `resign` action wasn't updated to pass the `version` param that was added in this commit https://github.com/fastlane/sigh/commit/dc9b49e15322eae7ee894a027075a597da1711f6. Working on a PR right now to fix!

**Comment by asfalcone** [comment]
@zachlipton Fixed in [1.66.0](https://github.com/fastlane/fastlane/releases/tag/1.66.0) :memo: :smiley:
